File: original/Code/Commando/csconsolecommandevent.cpp
#include "global.h"
#include "csconsolecommandevent.h"


#include "networkobjectfactory.h"
#include "gamemode.h"
#include "cnetwork.h"
#include "consolefunction.h"
#include "apppackettypes.h"
#include "networkobjectmgr.h"

DECLARE_NETWORKOBJECT_FACTORY(cCsConsoleCommandEvent, NETCLASSID_CSCONSOLECOMMANDEVENT);

//-----------------------------------------------------------------------------
cCsConsoleCommandEvent::cCsConsoleCommandEvent(void)
{
	::strcpy(Command, "");

	Set_App_Packet_Type(APPPACKETTYPE_CSCONSOLECOMMANDEVENT);
}

//-----------------------------------------------------------------------------
void
cCsConsoleCommandEvent::Init(LPCSTR command)
{

	::strcpy(Command, command);

	Set_Network_ID(NetworkObjectMgrClass::Get_New_Client_ID());

	Set_Object_Dirty_Bit(0, BIT_CREATION, true);
}

//-----------------------------------------------------------------------------
void
cCsConsoleCommandEvent::Act(void)
{

	if (GameModeManager::Find("Combat")->Is_Active()) {
		ConsoleFunctionManager::Parse_Input(Command);
	}
}

//-----------------------------------------------------------------------------
void
cCsConsoleCommandEvent::Export_Creation(BitStreamClass & packet)
{

	cNetEvent::Export_Creation(packet);

	packet.Add_Terminated_String(Command);

	Set_Delete_Pending();
}

//-----------------------------------------------------------------------------
void
cCsConsoleCommandEvent::Import_Creation(BitStreamClass & packet)
{

	cNetEvent::Import_Creation(packet);

	packet.Get_Terminated_String(Command, sizeof(Command));

	Act();

	Set_Delete_Pending();
}

File: kotlin-server/server/src/main/kotlin/ccr/server/net/CsConsoleCommandEvent.kt
package ccr.server.net

import ccr.net.bitstream.BitStream
import ccr.server.Network

// C++: cCsConsoleCommandEvent — networkClassId = NETCLASSID_CSCONSOLECOMMANDEVENT = 1035
// Client→Server event carrying a console command string from the client.
// Wire format (BIT_CREATION):
//   Command (terminatedString, ASCII, permitEmpty=false)
class CsConsoleCommandEvent(
    var command: String = "",
) : NetEvent() {
    override val networkClassId: Int = 1035

    override fun exportCreation(packet: BitStream) {
        packet.addTerminatedString(command)
    }

    override fun importCreation(packet: BitStream) {
        command = packet.getTerminatedString()
        actIfWiredUp()
    }

    override fun act() {
        println("[GAME] CSCONSOLECOMMANDEVENT command=$command (no console dispatch)")
        setDeletePending()
    }
}
